sdk/samples/core/mosquitto_multiple_rpc_sample.c

```diff
@@ -0,0 +1,603 @@
+{
+  // TODO: Payload should be generated and serialized
+  return invoke_begin(
+      start_module_command_name,
+      AZ_SPAN_FROM_STR("{\"RequestTimestamp\":1691530585198,\"RequestedFrom\":\"mobile-app\"}"));
+}
+
+az_result invoke_stop_module()
+{
+  // TODO: Payload should be generated and serialized
+  return invoke_begin(
+      stop_module_command_name,
+      AZ_SPAN_FROM_STR("{\"RequestTimestamp\":1691530585198,\"RequestedFrom\":\"mobile-app\"}"));
+}
+
+int main(int argc, char* argv[])
+{
+  (void)argc;
+  (void)argv;
+
+  /* Required before calling other mosquitto functions */
+  if (mosquitto_lib_init() != MOSQ_ERR_SUCCESS)
+  {
+    printf(LOG_APP "Failed to initialize MosquittoLib\n");
+    return -1;
+  }
+
+  printf(LOG_APP "Using MosquittoLib version %d\n", mosquitto_lib_version(NULL, NULL, NULL));
+
+  az_log_set_message_callback(az_sdk_log_callback);
+  az_log_set_classification_filter_callback(az_sdk_log_filter_callback);
+
+  connection_context = az_context_create_with_expiration(
+      &az_context_application, az_context_get_expiration(&az_context_application));
+
+  az_mqtt5 mqtt5;
+  struct mosquitto* mosq = NULL;
+
+  LOG_AND_EXIT_IF_FAILED(az_mqtt5_init(&mqtt5, &mosq, NULL));
+
+  az_mqtt5_x509_client_certificate primary_credential = (az_mqtt5_x509_client_certificate){
+    .cert = cert_path1,
+    .key = key_path1,
+  };
+
+  az_mqtt5_connection_options connection_options = az_mqtt5_connection_options_default();
+  connection_options.client_id_buffer = client_id;
+  connection_options.username_buffer = username;
+  connection_options.password_buffer = AZ_SPAN_EMPTY;
+  connection_options.hostname = hostname;
+  connection_options.client_certificates[0] = primary_credential;
+
+  LOG_AND_EXIT_IF_FAILED(az_mqtt5_connection_init(
+      &mqtt_connection, &connection_context, &mqtt5, mqtt_callback, &connection_options));
+
+  LOG_AND_EXIT_IF_FAILED(az_platform_mutex_init(&pending_server_command_mutex));
+  LOG_AND_EXIT_IF_FAILED(az_platform_mutex_acquire(&pending_server_command_mutex));
+
+  pending_server_command.request_data = AZ_SPAN_FROM_BUFFER(server_request_payload_buffer);
+  pending_server_command.content_type = AZ_SPAN_FROM_BUFFER(server_content_type_buffer);
+  pending_server_command.correlation_id = AZ_SPAN_EMPTY;
+  pending_server_command.response_topic = AZ_SPAN_FROM_BUFFER(server_response_topic_buffer);
+  pending_server_command.request_topic = AZ_SPAN_FROM_BUFFER(server_request_topic_buffer);
+
+  LOG_AND_EXIT_IF_FAILED(az_platform_mutex_release(&pending_server_command_mutex));
+
+  az_mqtt5_property_bag server_property_bag;
+  mosquitto_property* server_mosq_prop = NULL;
+  LOG_AND_EXIT_IF_FAILED(
+      az_mqtt5_property_bag_init(&server_property_bag, &mqtt5, &server_mosq_prop));
+
+  az_mqtt5_rpc_server_options server_options = az_mqtt5_rpc_server_options_default();
+  server_options.subscription_topic_format = server_subscription_topic_format;
+
+  LOG_AND_EXIT_IF_FAILED(az_rpc_server_policy_init(
```

Comment:
Policies shouldn't leak through the public API.
Could we rename `az_mqtt5_rpc_server_policy` to `az_mqtt5_rpc_server` and the current `az_mqtt5_rpc_server` to `az_mqtt5_rpc_server_codec`? (same for client)

Reply:
Will address in follow-up PR